torch/utils/data/datasets/routeddecoderdataset.py

```diff
@@ -0,0 +1,46 @@
+from torch.utils.data.dataset import IterableDataset
+from torch.utils.data.datasets.decoder import (
+    Decoder,
+    basichandlers as decoder_basichandlers,
+    imagehandler as decoder_imagehandler)
+
+from typing import Iterable, Iterator, Union, List
+
+class RoutedDecoderIterableDataset(IterableDataset):
+    r""" :class:`RoutedDecoderIterableDataset`.
+
+    IterableDataset to decode binary streams from input iterables,
+    yield pathname and decoded binary stream in a tuple.
+    args:
+        dataset: Iterable dataset that provides pathname and binary stream in tuples
+        decoders: user defined decoders, if None, basic and image decoders will be set as default
+        length: a nominal length of the dataset
+    """
+
+    def __init__(
+            self,
+            dataset : Iterable,
+            *,
+            decoders : Union[None, List[str]] = None,
```

Comment:
It seems the wrong type hint. Can we directly use `Decoder` as input?

Reply:
lol, why I put a str there :), should be callables.